src/Symfony/Security/State/AccessCheckerProvider.php

```diff
@@ -59,15 +59,15 @@ public function provide(Operation $operation, array $uriVariables = [], array $c
                 $message = $operation->getSecurityMessage();
         }
 
-        $body = $this->decorated->provide($operation, $uriVariables, $context);
-        if (null === $isGranted) {
-            return $body;
+        if (
+            null === $isGranted
+            // On a GraphQl QueryCollection we want to perform security stage only on the top-level query
+            || ($operation instanceof QueryCollection && null !== ($context['source'] ?? null))
+        ) {
+            return $this->decorated->provide($operation, $uriVariables, $context);
         }
 
-        // On a GraphQl QueryCollection we want to perform security stage only on the top-level query
-        if ($operation instanceof QueryCollection && null !== ($context['source'] ?? null)) {
-            return $body;
-        }
+        $body = 'pre_read' === $this->event ? null : $this->decorated->provide($operation, $uriVariables, $context);
```

Comment:
```suggestion
        $preRead = 'pre_read' === $this->event;
        $body = $preRead ? null : $this->decorated->provide($operation, $uriVariables, $context);
```

Nit: micro-optimization, to not do the string comparison many times and improve readability.